### PR TITLE
Add deployment API

### PR DESCRIPTION
**Pull request by Lorac**

### CODE REVIEW

File: deployments.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		return nil, resp, err
+	}
+
+	return ds, resp, err
+}
+
+// GetProjectDeployment get a deployment for a project.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/deployments.html#get-a-specific-deployment
+func (s *DeploymentsService) GetProjectDeployment(pid interface{}, deploymentID int, options ...OptionFunc) (*Deployment, *Response, error) {
```
Comment (svanharmelen) [inline deployments.go]: I tent to call `deploymentID`, just `deployment` in most other API calls...

File: deployments.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		} `json:"pipeline"`
+		Runner *Runner `json:"runner"`
+	} `json:"deployable"`
+}
+
+// ListDeploymentsOptions represents the available ListProjectDeployments() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/deployments.html#list-project-deployments
+type ListDeploymentsOptions struct {
```
Comment (svanharmelen) [inline deployments.go]: I know it gets a bit long, but following the conventions this should be called `ListProjectDeploymentsOptions`...

**Comment by Lorac** [comment]
Pushed fixes

File: deployments.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+// ListProjectDeploymentsOptions represents the available ListProjectDeployments() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/deployments.html#list-project-deployments
+type ListProjectDeploymentsOptions struct {
+	ListOptions
+	OrderBy *OrderByValue `json:"order_by,omitempty"`
+	Sort    *string       `json:"sort,omitempty"`
```
Comment (svanharmelen) [inline deployments.go]: Hmm... I've seem to have missed it, but options struct need both `url` and `json` tags. If you can add those, this one can be merged as well 👍

Comment (Lorac) [reply deployments.go]: Oh right!

**Comment by svanharmelen** [comment]
Here we go, LGTM! 👍

**Pull request merged**